Removed manual setting of date to the first of the month

nathandines opened the pull request:
This setting seemed unnecessary and was causing issues for people in GMT+ timezones. Current month data was displaying as the previous month due to me being in GMT+11.

Issue only occurred when showing estimated data usage (didn't appear when using "--style 0")

nathandines commented:
actually, looks like there might be an issue with this. I'm going to do some further digging before submitting another pull

nathandines commented:
Upon second look, it seems like the work was fine. Would be great if you could review and pull it into the main repo.

vergoh commented:
I'll have to admit, I have some difficulties recalling why I implemented the mosecs() function like that (in version 1.8) and why did I come up with such bad function name. However, as far as I can tell, only the traffic estimates and rates should get affected, not the data itself. Could you provide an example output of what you see without the modification compared to what you are expecting?

nathandines commented:
Please find below example outputs of both the patched version, and the unpatched version (represented by a comment on the command line).

I'll say from the offset that I'm no C expert. When I was debugging this issue, I found that the function "mosecs" was somehow changing the value of "d" in the function "showshort" in dbshow.c; before lines 294 and 295, "d" was 1444619402, and after those lines it had changed to 1443618026 (which isn't even a round minute!). I found that it was the two lines which I removed which was doing it, and I don't believe they are necessary, so I removed them. It would be good to know why the variable was being changed outside the local scope (and to a number which wasn't even a round minute as is to be expected)... any ideas? Even when I changed the variable name of "d" within mosecs to "dd", the issue still persisted, so I really have no clue how this was happening.
![selection_001](https://cloud.githubusercontent.com/assets/793167/10500367/65e548da-7323-11e5-87ac-f0987b8919ae.png)

vergoh commented:
I found an explanation from the ctime man page:
> The  localtime()  function  converts  the calendar time timep to broken-down time representation, expressed relative to the user's specified timezone.  The function acts as if it called tzset(3) and sets the  external  variables  tzname  with  information  about the current timezone, timezone with the difference between Coordinated Universal Time (UTC) and local standard time in seconds, and daylight to  a  nonzero  value  if  daylight  savings  time  rules apply during some part of the year.  The return value points to a statically allocated struct which might be overwritten by subsequent calls to any of the date and time functions.  The localtime_r()  function  does  the  same,  but  stores the data in a user-supplied struct.  It need not set tzname, timezone, and daylight.

The relevant detail that I had missed is
> The return value points to a statically allocated struct which might be overwritten by subsequent calls to any of the date and time functions.

That's why any use of the struct tm pointer after mosecs() had been called contained something that wasn't expected to be there resulting in wrong dates to be shown. I wrote two testcases for checking the problem and noticed that also the return value of mosecs() could be off by several hours in seconds due to the used timezone not having been taken into account.

vergoh commented:
67cfc079feb3d7579b211f4356d5fd3974812895 fixes this issue.

nathandines commented:
That's great. Your solution is far more elegant.

Thanks for looking into this and providing such a fast turnaround :)
